pennylane/pauli/dla/center.py

```diff
@@ -54,25 +57,37 @@ def center(
     In this case, just ``X(0)``.
 
     >>> qml.center(g)
-    [X(0)]
+    [-0.9999999999999999 * X(0)]
 
     """
+    if len(g) < 2:
+        # A length-zero list has zero center, a length-one list has full center
+        return g
+
+    adjoint_repr = structure_constants(g, pauli)
+    # Compute kernel for adjoint rep of each DLA element
+    kernels = [null_space(ad_x) for ad_x in adjoint_repr]
+    # If any kernels vanishes, their overlap vanishes as well
+    if any(k.shape[1] == 0 for k in kernels):
+        return []
+    # Compute the complements of the kernels
+    supports = [null_space(k.T) for k in kernels]
+    # Combine all complements
+    combined_support = np.hstack(supports)
+    if combined_support.shape[1] == 0:
+        return g
+    # Compute the complement of the combined complements: It is the intersection of the kernels
+    center_coefficients = null_space(combined_support.T)
+
+    # Construct operators from numerical output and convert to desired format
+    res = [sum(c * x for c, x in zip(c_coeffs, g)) for c_coeffs in center_coefficients.T]
+
+    have_paulis = isinstance(g[0], (PauliWord, PauliSentence))
```

Comment:
instead of this I would suggest to convert everything to paulis if necessary in the beginning, e.g. the following like in `lie_closure`

```
    if not all(isinstance(op, (PauliSentence, PauliWord)) for op in generators):
        if pauli:
            raise TypeError(
                "All generators need to be of type PauliSentence or PauliWord when using pauli=True in lie_closure."
            )

        generators = [
            rep if (rep := op.pauli_rep) is not None else qml.pauli.pauli_sentence(op)
            for op in generators
        ]
```

Reply:
But we don't need Paulis for this function I think. So for `pauli=False` we would just make Paulis to make them into operations later :sweat_smile:

Reply:
Oh yeah good point, this just looked odd at first but I think it covers all cases correctly. You could trick it by providing a mixed list though, e.g. `[PauliWord({0: "X"}), X(1), Y(0) @ Y(1)]`

Reply:
Done.